zerver/actions/user_topics.py

```diff
@@ -19,22 +19,25 @@ def do_set_user_topic_visibility_policy(
     *,
     visibility_policy: int,
     last_updated: Optional[datetime.datetime] = None,
-    ignore_duplicate: bool = False,
     skip_muted_topics_event: bool = False,
 ) -> None:
     if last_updated is None:
         last_updated = timezone_now()
 
-    set_user_topic_visibility_policy_in_database(
+    skip_send_event = set_user_topic_visibility_policy_in_database(
         user_profile,
         stream.id,
         topic,
         visibility_policy=visibility_policy,
         recipient_id=stream.recipient_id,
         last_updated=last_updated,
-        ignore_duplicate=ignore_duplicate,
     )
 
+    # If there are duplicate requests to update the visibility_policy,
+    # don't send duplicate events.
+    if skip_send_event:
+        return
```

Comment:
As noted above, I think this will be a lot more readable if we have the above lines be `database_changed = set_user_topic_visibility_policy_in_database(...)` and then this is `if not database_changed: return` -- since that expresses clearly the logic that we're only sending an event if we changed something.

(And then I think you don't need the comment explaining the reasoning)